packages/sanity/src/_internal/cli/threads/validateDocuments.ts

```diff
@@ -24,6 +24,7 @@ import {isRecord, validateDocument} from 'sanity'
 const MAX_VALIDATION_CONCURRENCY = 100
 const DOCUMENT_VALIDATION_TIMEOUT = 30000
 const REFERENCE_INTEGRITY_BATCH_SIZE = 100
+const EXCLUDED_DOCUMENT_TYPE_PREFIXES = ['system.', 'sanity.']
```

Comment:
I wonder whether there might ever be a reason to validate assets:

1. They can be modified using the API, so can end up in an invalid state. CLDX-1377 covers a scenario like this. Although there are no existing validation rules that would have caught this problem, it's a real issue that affected users.
2. In the future, it may be possible for users to extend these schemas, likely making bulk validating them more useful.

Reply:
> They can be modified using the API, so can end up in an invalid state. CLDX-1377 covers a scenario like this. Although there are no existing validation rules that would have caught this problem, it's a real issue that affected users.

The decision to exclude `sanity.` documents came from @runeb but I do see your point. Any built-in `sanity.` type can happy be validated if we have a schema definition. The built-in ones are [here](https://github.com/sanity-io/sanity/blob/9b3eae9781b1a33e0e7a9845e107d2c73eaf5437/packages/sanity/src/core/schema/createSchema.ts#L19-L32).

@runeb what do you think? should we allow validating the built-in types too?

> In the future, it may be possible for users to extend these schemas, likely making bulk validating them more useful.

If a user does extend any of the built-in schemas, that should get a new document `_type` and therefore would then get included in the validation results. This line just excludes `_type`s that start with `sanity.*` or `system.*`

Reply:
Just followed up with @runeb and we decided to keep validating sanity built-in types so I removed the `sanity.*` prefix but there's still some things to discuss. I tagged you in slack.

Reply:
Sounds good!

> If a user does extend any of the built-in schemas, that should get a new document _type and therefore would then get included in the validation results.

Sorry, I probably phrased that badly. By _extend_, I meant adding fields to the system type, e.g. adding a `license` field directly to `sanity.imageAsset`.